File: day 26/nested_logic.py
"""Get the fine on a returned library book."""

def get_fine(expected, returned):
    """Use layered logic to find the fine due."""
    if expected[2] < returned[2]:#pylint:disable=no-else-return
        return 10000
    elif expected[2] > returned[2] or (expected[0] >= returned[0] and expected[1] >= returned[1]):
        return 0
    elif expected[0] < returned[0] and expected[1] == returned[1]:
        return 15 * (returned[0] - expected[0])
    return 500 * (returned[1] - expected[1])

if __name__ == "__main__":
    RETURNED = [int(num) for num in input().split()]
    EXPECTED = [int(num) for num in input().split()]
    print(get_fine(EXPECTED, RETURNED))

File: day 26/nested_logic.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <sstream>
using namespace std;

int* getDate()
{ 
    string str_date1;
    getline(cin, str_date1);
    // Used to split string around spaces. 
    istringstream ss(str_date1);
    int *dates = new int[3];
    int index = 0;
    // Traverse through all words 
    do { 
        // Read a word 
        int num; 
        ss >> num; 
        dates[index] = num;
        index++;
        // While there is more to read 
    } while (ss); 
    return dates;
} 

int getFine(int* returned, int* expected){
    if (expected[2] < returned[2])
        return 10000;
    else if ((expected[0] >= returned[0] && expected[1] >= returned[1])
        || expected[2] > returned[2]){
        return 0;
    }
    else if (expected[0] < returned[0] && expected[1] == returned[1])
        return 15*(returned[0]-expected[0]);
    else {
        return 500*(returned[1]-expected[1]);
    }
}

int main() {

    int* returned = getDate();
    int* expected = getDate();
    cout << getFine(returned, expected) << endl;

    return 0;
}
